internal/cmd/ske/credentials/kubeconfig/login/login.go

```diff
@@ -0,0 +1,223 @@
+			return nil
+		},
+	}
+	return cmd
+}
+
+func parseInput() (*inputModel, error) {
+	obj, _, err := exec.LoadExecCredentialFromEnv()
+	if err != nil {
+		return nil, fmt.Errorf("LoadExecCredentialFromEnv: %w", err)
+	}
+
+	if err := clientauthenticationv1.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+
+	obj, err = scheme.Scheme.ConvertToVersion(obj, clientauthenticationv1.SchemeGroupVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ConvertToVersion: %w", err)
+	}
+
+	execCredential, ok := obj.(*clientauthenticationv1.ExecCredential)
+	if !ok {
+		return nil, fmt.Errorf("Conversion to ExecCredential failed")
+	}
+	if execCredential == nil || execCredential.Spec.Cluster == nil {
+		return nil, fmt.Errorf("ExecCredential contains not all needed fields")
+	}
+	config := &SKEClusterConfig{}
+	err = json.Unmarshal(execCredential.Spec.Cluster.Config.Raw, config)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+
+	return &inputModel{
+		ClusterName: config.ClusterName,
+		ProjectId:   config.STACKITProjectID,
+		CacheKey:    fmt.Sprintf("ske-login-%x", sha256.Sum256([]byte(execCredential.Spec.Cluster.Server))),
+	}, nil
+}
+
+func buildRequest(ctx context.Context, apiClient *ske.APIClient, model *inputModel) ske.ApiCreateKubeconfigRequest {
+	req := apiClient.CreateKubeconfig(ctx, model.ProjectId, model.ClusterName)
+	expirationSeconds := "1800" // 30 min
+
+	return req.CreateKubeconfigPayload(ske.CreateKubeconfigPayload{ExpirationSeconds: &expirationSeconds})
+}
+
+func parseKubeConfigToExecCredential(kubeconfig *rest.Config) (*clientauthenticationv1.ExecCredential, error) {
+	certPem, _ := pem.Decode(kubeconfig.CertData)
+	if certPem == nil {
+		return nil, fmt.Errorf("login SKE kubeconfig")
+	}
+
+	certificate, err := x509.ParseCertificate(certPem.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("login SKE kubeconfig: %w", err)
+	}
+
+	outputExecCredential := clientauthenticationv1.ExecCredential{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: clientauthenticationv1.SchemeGroupVersion.String(),
+			Kind:       "ExecCredential",
+		},
+		Status: &clientauthenticationv1.ExecCredentialStatus{
+			ExpirationTimestamp:   &v1.Time{Time: certificate.NotAfter.Add(-time.Minute * 15)},
+			ClientCertificateData: string(kubeconfig.CertData),
+			ClientKeyData:         string(kubeconfig.KeyData),
+		},
+	}
+	return &outputExecCredential, nil
+}
+
+func getCachedKubeConfig(key string) *rest.Config {
+	cachedKubeconfig, err := cache.GetObject(key)
+	if err != nil {
+		return nil
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(cachedKubeconfig)
+	if err != nil {
+		return nil
+	}
+
+	return restConfig
+}
+
+type SKEClusterConfig struct {
+	STACKITProjectID string `json:"stackitProjectId"`
+	ClusterName      string `json:"clusterName"`
+}
+
+func getCacheAndOutputKubeconfig(ctx context.Context, cmd *cobra.Command, apiClient *ske.APIClient, model *inputModel, refresh bool, oldKubeconfig *rest.Config) error {
```

Comment:
I find the argument name `refresh` confusing. Since actually when `refresh` is true, you are returning the old kube config. 
So I would say it should be rather something like `getCachedKubeconfig`

Reply:
The name of the method is also misleading because from the first look I thought we are getting the cached kubeconfig, but actually we are getting, then caching, then output

Reply:
I've renamed the function and the `refresh` parameter. Let me know if it's better now.

Reply:
It's much nicer now :)